Automattic/pocket-casts-android
# Sync widget theme setting

## Description

One of the settings listed for sync project #1739. This PR adds syncing to the global settings for widget theme.

## Testing Instructions

> [!note]
> You need to test this with staging because changes aren't deployed to production, yet.
> 
> To test on staging use the `debug` variant instead of the `debugProd` one.

> [!note]
> You need to test with Android 12+ to have [Dynamic Color](https://developer.android.com/develop/ui/views/theming/dynamic-colors) enabled

1. Have two devices D1 and D2.
2. Make sure that both devices are synced before starting the test by refreshing apps in the profile.
3. D1: Set [Dynamic Color](https://developer.android.com/develop/ui/views/theming/dynamic-colors) to something visually obvious like blue.
4. D1: Add Pocket Casts widget to the home screen.
5. D2: Set [Dynamic Color](https://developer.android.com/develop/ui/views/theming/dynamic-colors) to something visually obvious like blue.
6. D2: Add Pocket Casts widget to the home screen.
7. D1: Go to `Profile`/`Settings (cog icon)`/`Appearance`.
8. D1: Toggle `Use dynamic colors for widget` setting `ON`.
9. D1: Minimize the app
10. D1: Notice that the dynamic color is applied to the widget.
11. D1: Go back to the app.
12. D1: Sync the device in the `Profile`.
13. D2: Sync the device in the `Profile`.
14. D2: Minimize the app.
15. D2: Notice that the dynamic color is applied to the widget.
16. D2: Go back to the app.
17. D2: Go to `Profile`/`Settings (cog icon)`/`Appearance`.
18. D2: Toggle `Use dynamic colors for widget` setting `OFF`.
19. D2: Sync the device in the `Profile`.
20. D1: Sync the device in the `Profile`.
21. D1: Minimize the app.
22. D1: Notice that if the dynamic color is not applied to the widget.

## Checklist
 
#### I have tested any UI changes...
